org-code-javabuilder/lib/src/main/java/org/code/javabuilder/OutputRedirectionStream.java

```diff
@@ -0,0 +1,74 @@
+package org.code.javabuilder;
+
+import java.io.OutputStream;
+
+/**
+ * An OutputStream that passes output to an OutputAdapter. It is intended to redirect output from
+ * the user program that is intended for the console. See
+ * https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html for full implementation
+ * details and contract.
+ */
+public class OutputRedirectionStream extends OutputStream {
+  private final OutputAdapter outputAdapter;
+  private final StringBuilder buffer;
+
+  public OutputRedirectionStream(OutputAdapter outputAdapter) {
+    super();
+    this.outputAdapter = outputAdapter;
+    this.buffer = new StringBuilder();
+  }
+  /**
+   * See: https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-- Checks the queue
+   * for existing bytes. If the queue is empty, polls the inputAdapter for new data. This is a
+   * blocking call.
+   *
+   * @return the first byte in the queue
+   */
+  //
```

Comment:
missing a function here? Or do we need to remove this comment?

Reply:
Haha oops. Thank you.